core/modules/layout/layout.flexible.inc

```diff
@@ -0,0 +1,1118 @@
+<?php
+/**
+ * @file
+ * Administrative functions for Layout module.
+ *
+ * This provides the UI to list, create, edit and delete layouts.
+ */
+
+/**
+ * Form to add a row to a flexible template.
+ *
+ * @param $flexible_template_id
+ *   The template to which a row is being added.
+ * @param $original_row
+ *   The row above or below which a new row is being inserted.
+ * @param $position
+ *   Where the new row is being inserted.
+ *
+ * @ingroup forms
+ */
+function layout_flexible_template_settings_form($form, &$form_state, $flexible_template_id = NULL) {
+  $form_state['template_data'] = layout_flexible_template_template($flexible_template_id);
+  $template_data = $form_state['template_data'];
+
+  $form['name'] = array(
+    '#title' => t('Template name'),
+    '#type' => 'textfield',
+    '#maxlength' => 128,
+    '#default_value' => $template_data['title'],
+    '#required' => TRUE,
+  );
+  $form['machine_name'] = array(
+    '#type' => 'machine_name',
+    '#maxlength' => 21,
+    '#default_value' => $template_data['name'],
+    '#machine_name' => array(
+      'exists' => 'layout_flexible_template_name_exists',
```

Comment:
Looks like in other places we just use the loader function instead of making a wrapper for `_exists()`.

```suggestion
      'exists' => 'layout_flexible_template_load',
```

Reply:
Done